include/seastar/core/metrics_api.hh

```diff
@@ -381,6 +443,109 @@ std::unique_ptr<metric_groups_def> create_metric_groups();
 
 }
 
+/*!
+ * \brief a relabel_config allows changing metrics labels dynamically
+ *
+ * The logic is similar to Prometheus configuration
+ * This is how Prometheus entry looks like:
+ *  - source_labels: [version]
+      regex: '([0-9]+\.[0-9]+)(\.?[0-9]*).*'
+      replacement: '$1$2'
+      target_label: svr
+ */
+class relabel_config {
+public:
```

Comment:
Let's move this into its own header (and use forward declarations) so that not everyone needs to pull \<regex>.

Reply:
ok, do note that metrics_api.hh is a header file that only a few files include, it is the api to control the metrics layer, prometheus for example uses it.

Reply:
Ah, no need then.